thrust/thrust/detail/alignment.h

```diff
@@ -157,10 +157,23 @@ struct aligned_type;
 /// \p aligned_storage provides the nested type `type`, which is a trivial type
 /// suitable for use as uninitialized storage for any object whose size is at
 /// most `Len` bytes and whose alignment requirement is a divisor of `Align`.
-/// 
+///
 /// The behavior is undefined if `Len` is 0 or `Align` is not a power of 2.
 ///
 /// It is an implementation of C++11's \p std::aligned_storage.
+///
+/// It has been deprecated with C++23.
+
+#if THRUST_CPP_DIALECT > 2020 // aligned_storage has been deprecated in c++23
+#if (THRUST_HOST_COMPILER == THRUST_HOST_COMPILER_CLANG)
+#pragma clang diagnostic push
+#pragma clang diagnostic clang diagnostic ignored "-Wdeprecated-declarations"
+#elif (THRUST_HOST_COMPILER == THRUST_HOST_COMPILER_GCC)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic clang diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
Should this be something along the following lines instead?
```suggestion
#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```